heroku3/models/app.py

```diff
@@ -203,12 +203,15 @@ def domains(self, **kwargs):
         """The domains for this app."""
         return self._h._get_resources(resource=("apps", self.name, "domains"), obj=Domain, app=self, **kwargs)
 
-    def add_domain(self, hostname):
+    def add_domain(self, hostname, sni_endpoint):
```

Comment:
@colinnutrislice should this be `hostname, sni_endpoint=None` that way when we upgrade our libraries, we will get the default behavior (the API will send null) without changing our invocation or adding arguments. 

Per https://devcenter.heroku.com/changelog-items/1938

Reply:
Sadly, I do not have permissions to create a branch in heroku3.

Also @tomusher this sni_endpoint PR.

Reply:
That was my initial thought as well.

I think there is an argument in having it not default to `None`, because the functionality is not the same as before.

Previously, with no `sni_endpoint`, it would figure out the appropriate SNI to use.

Now, it won't attach the domain to any SNI if `None `is passed though.

If someone was relying on the endpoint previously figuring out which SNI to use, it will now start to add domains with no SNI attached at all and perhaps it's good to have this be a breaking change as an indicator to others?

I'm good with either and honestly don't really have the authority to make the call.

Reply:
From Heroku's announcement:
`Beginning October 31, 2020, API requests to the Domain Create endpoint will accept an additional sni_endpoint parameter.`

No heroku3 (this repo) library users have started using sni_endpoint, because we have yet to cut and release this commit to PyPI. 
So no one relying on heroku3 will be broken by setting the default value of sni_endpoint=None.

Go ahead and set to None, it will be of most benefit to the community.  Most users will not have a need for sni_endpoint right away, and if they do, they can always pass in the value they want to the method.

Reply:
That's not quite what I'm saying.

![add_domain](https://user-images.githubusercontent.com/44236794/139492183-84351390-2f98-45a6-b2da-01a238fc3d33.png)

The top one was added with the most recent release, 4.2.3:
`app.add_domain('a1-testadd.example.com')`

The bottom one was added with the last code here, passing `None` in:
`app.add_domain('a2-testadd.example.com', None)`

So with your proposed change and a new release, the endpoint starts to go from providing no `sni_endpoint` to providing it with `None` and the functionality changes from automatically figuring out the appropriate SNI certificate to attaching no SNI certificate at all.

If we don't have it default to `None` it is an indicator to the community that the functionality is changing. Otherwise, it slips in and suddenly domains are being added with no SNI.

Reply:
The python requests library converts None to null.  
Take a look at:
https://stackoverflow.com/a/36271843
"Setting a dictionary element to None is how you explicitly say that you don't want that parameter to be sent to the server."

Reply:
Oh I see what you're saying.  The SSL cert was actually not created...

Reply:
Correct. Not attached to the domain, because None/null was sent along for the newly required `sni_endpoint` param.

Reply:
It's also the reason why this PR adds `cert_domains`, so that we can find the appropriate SNI ourselves and pass that up with `add_domain`.

Reply:
Yeah, we're going to need to explain that to heorku3 users - we'll need to [update the heroku3/README.rst](https://github.com/martyzz1/heroku3.py/blob/12591ef8d825b8bb53616538626a910957c525ab/README.rst) with that gotcha
That passing sni_endpoint=None will result in no SSL cert being created
and that passing snl_endpoint=cert_domains will pull the domain SSL cert. 

And we should note that an additional parameter is needed on add_domain(

Reply:
https://github.com/martyzz1/heroku3.py/pull/113

Reply:
> and that passing snl_endpoint=cert_domains will pull the domain SSL cert.

that part's not accurate. Have to do some additional work to scan through `app.sni_endpoints()` -> `ssl_cert` -> `cert_domains`. I added an example to the README.